src/configuration.ts

```diff
@@ -0,0 +1,332 @@
+    flags,
+    bandits,
+    precomputed,
+  }: {
+    flags?: FlagsConfig;
+    bandits?: BanditsConfig;
+    precomputed?: PrecomputedConfig;
+  }): Configuration {
+    return new Configuration({ flags, bandits, precomputed });
+  }
+
+  /**
+   * Initializes a Configuration from a "configuration wire" format (this is the format returned by
+   * `toString`).
+   *
+   * @public
+   */
+  public static fromString(configurationWire: string): Configuration {
+    // TODO: we're assuming that `configurationWire` is properly formatted.
+    const wire: ConfigurationWire = JSON.parse(configurationWire);
+
+    let flags: FlagsConfig | undefined;
+    let bandits: BanditsConfig | undefined;
+    let precomputed: PrecomputedConfig | undefined;
+
+    if (wire.config) {
+      flags = {
+        response: JSON.parse(wire.config.response),
+        etag: wire.config.etag,
+        fetchedAt: wire.config.fetchedAt,
+      };
+    }
+
+    if (wire.bandits) {
+      bandits = {
+        response: JSON.parse(wire.bandits.response),
+        etag: wire.bandits.etag,
+        fetchedAt: wire.bandits.fetchedAt,
+      };
+    }
+
+    if (wire.precomputed) {
+      precomputed = {
+        response: JSON.parse(wire.precomputed.response),
+        etag: wire.precomputed.etag,
+        fetchedAt: wire.precomputed.fetchedAt,
+        subjectKey: wire.precomputed.subjectKey,
+        subjectAttributes: wire.precomputed.subjectAttributes,
+        banditActions: wire.precomputed.banditActions,
+      };
+    }
+
+    return new Configuration({
+      flags,
+      bandits,
+      precomputed,
+    });
+  }
+
+  /** Serializes configuration to "configuration wire" format. */
+  public toString(): string {
+    const wire: ConfigurationWire = {
+      version: 1,
+    };
+    if (this.parts.flags) {
+      wire.config = {
+        ...this.parts.flags,
+        response: JSON.stringify(this.parts.flags.response),
+      };
+    }
+    if (this.parts.bandits) {
+      wire.bandits = {
+        ...this.parts.bandits,
+        response: JSON.stringify(this.parts.bandits.response),
+      };
+    }
+    if (this.parts.precomputed) {
+      wire.precomputed = {
+        ...this.parts.precomputed,
+        response: JSON.stringify(this.parts.precomputed.response),
+      };
+    }
+    return JSON.stringify(wire);
+  }
+
+  /**
+   * Returns a list of known flag keys (for debugging purposes).
+   *
+   * If underlying flags configuration is obfuscated, the returned
+   * flag values will be obfuscated as well.
+   */
+  public getFlagKeys(): string[] {
+    if (this.parts.flags) {
+      return Object.keys(this.parts.flags.response.flags);
+    }
+    if (this.parts.precomputed) {
+      return Object.keys(this.parts.precomputed.response.flags);
+    }
+    return [];
+  }
+
+  /** @internal */
+  public getFlagsConfiguration(): FlagsConfig | undefined {
+    return this.parts.flags;
+  }
+
+  /** @internal */
+  public getFetchedAt(): Date | undefined {
```

Comment:
There are two timestamps we care about _published_ and _fetched_. This seems to handle the fetched one, but what about publish time? (pulled from createdAt in the response I reckon)

Reply:
it's never used 🤷